classes/Product.php

```diff
@@ -3695,6 +3723,7 @@ public static function getQuantity(
         Cart $cart = null,
         $idCustomization = null
     ) {
+        // @todo: check whether getAvailableQuantityInStock() can be used
         if (Pack::isPack((int) $idProduct)) {
             return Pack::getQuantity($idProduct, $idProductAttribute, $cacheIsPack, $cart, $idCustomization);
```

Comment:
This line is very strange 🤔 . If you check the code below, you see that `Product:: getQuantity()` behavior is
1) get the product available items in stock and 2) remove the items in the cart from it.

So if product X has stock of 10, and in your cart you already added 5 items of product X, then `Product:: getQuantity()` should return 5. Cart items are decreased.

Buuuuut ... according to line 3727, for packs we just skip step 2 😱 . @marionf do you know whether this is intentional ?

### So it seems the following scenario is currently active ⬇️ 

On my shop, I seel 10 units of my beautiful black tshirt !
Customer John Doe comes, and adds 5 units of the black tshirt in its cart => cool.
Oh ? customer John Doe then adds 6 units of the black tshirt in its cart => I tell him he cant do that, he wants to order 11 units but I only have 10.

The day after, I also sell a black hat. I have 10 units of it. So I create a pack "black tshirt + black hat" !
Customer John Doe comes (again ! I love this customer), and adds 5 units of the pack in its cart => cool.
Oh ? customer John Doe then adds 6 units of the pack in its cart => I tell him he can, go ahead man ! 😄

Reply:
If we find out this is another bug, then I'll be able to implement `@todo: check whether getAvailableQuantityInStock() can be used`